src/argv.c

```diff
@@ -315,6 +315,29 @@ format_expand_arg(struct format_context *format, const char *name, const char *e
 			return false;
 		return string_format_from(format->buf, &format->bufpos, "%s", value);
 	}
+	if (!prefixcmp(name, "%(sh")) {
+		char cbuf[SIZEOF_STR];
+		const char *c="";
+		char value[SIZEOF_STR]={0};
+		const char *cstart = name + STRING_SIZE("%(sh");
+		const int clen = end - cstart - 1;
+		int size;
+		FILE *cp;
+		if (end && clen > 0 && string_format(cbuf, "sh -c '%.*s'", clen, cstart)) {
+			c=cbuf;
+			while (isspace(*c))
+				c++;
+		}
+		if (!*c || strlen(c)==8)
+			return false;
+		cp=popen(c,"r");
+		size=fread(value,1,SIZEOF_STR-1,cp);
+		if (value[0]==0 || size==0)
+			return false;
+		if (value[size-1]=='\n')
+			value[size-1]='\0';
+		return string_format_from(format->buf, &format->bufpos, "%s", value);
```

Comment:
the entire output becomes one string argument.
This is good for some cases but if we want to compute multiple arguments for a tig command, it becomes awkward.

An `eval` command might remedy that but I'm not sure if this is a good design, see also https://en.wikipedia.org/wiki/Inner-platform_effect

Maybe instead of

	bind generic T :goto '%(sh git rev-list %(commit)~..|head -1)'

we should write

	bind generic T %echo goto "$(git rev-list "$tig_commit"~..|head -1)"

where the % is the magic key for "run this shell command and run the output as tig command"

Something like this seems like a reasonably general approach but I'm sure there are many things I didn't consider.
Sadly I don't have much time these days.

Reply:
Do you mean that it's not possible to embed `%(commit(` and others inside `%(sh …)`? When you write: `the entire output becomes one string argument.`? Or something else? Because I've implemented support for this in second patch 4301605b.